dissect/target/filesystems/tar.py

```diff
@@ -156,7 +156,7 @@ def lstat(self) -> fsutil.stat_result:
 
 
 class TarFilesystemDirectoryEntry(VirtualDirectory):
-    def __init__(self, fs: TarFilesystem, path: str, entry: tarfile.TarInfo):
+    def __init__(self, fs: VirtualFilesystem, path: str, entry: tarfile.TarInfo):
```

Comment:
This change seems incorrect, as these entries are only created by instances of `TarFilesystem` and it will pass `self` into `fs`.

Reply:
Yeah sorry this was from a different fix approach.